### PR TITLE
Adds shared lib to check whether a doc is a valid registration

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

The check is similar, albeit more simplistic, to the one in `registration` Sentinel transition.
Compensates for the lack of `transition` meta data present in the document. 

medic/medic-webapp#4806
# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Hey @garethbowen, please review.

Alternatively, I can make the view emit a `true` value if `doc.transitions.registration.ok` is `true` and use that to validate registrations in the `create-patient-contacts` migration (assuming that it will always run on old data which has saved transitions ?). Thoughts?
 
Also, I wanted to avoid creating a new shared lib and initially added it to `message-utils`, but that was forcing some extra unused dependencies  to be added to `api`.

Thanks for your time!

File: shared-libs/registration-utils/src/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      (doc.errors && doc.errors.length) ||
+      !settings ||
+      !settings.registrations ||
+      doc.type !== 'data_record' ||
+      !doc.form) {
+    return false;
+  }
+
+  var registrationConfiguration = settings.registrations.find(function(conf) {
+    return new RegExp('^\W*' + conf.form + '\\W*$','i').test(doc.form);
```
Comment (garethbowen) [inline shared-libs/registration-utils/src/index.js]: I know this is how the old code was but it seems really wasteful to create a regex for each configured registration every time this is called. Instead can we trim and lowercase the doc.form and just compare with the configured string?

File: shared-libs/registration-utils/src/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  if (!registrationConfiguration) {
+    return false;
+  }
+
+  var form = settings.forms && settings.forms[doc.form];
+  return Boolean(
+    doc.content_type === 'xml' ||
+    (form && form.public_form) ||
+    (form && doc.contact)
+  );
```
Comment (garethbowen) [inline shared-libs/registration-utils/src/index.js]: I think this section could be split up for readability but also for slight performance gains, eg:

```js
  if (doc.content_type === 'xml') {
    return true;
  }

  var form = settings.forms && settings.forms[doc.form];

  if (!form) {
    return false;
  }
  return Boolean(form.public_form || doc.contact);
```

Some of those lines could use comments.

File: shared-libs/registration-utils/test/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
+var sinon = require('sinon').sandbox.create(),
```
Comment (garethbowen) [inline shared-libs/registration-utils/test/index.spec.js]: The tests can be ES6, right? So `const`, `let`, arrow functions, etc.

**Comment by garethbowen** [comment]
I'm not sure if this is ready for review but I've reviewed it anyway and it looks good to me! Nice one.

**Pull request merged**